cnxepub/adapters.py

```diff
@@ -347,35 +347,61 @@ def adapt_single_html(html):
     return binder
 
 
-def _adapt_single_html_tree(parent, elem, nav_tree, pages_by_id=None, depth=0):
+def _adapt_single_html_tree(parent, elem, nav_tree, id_map=None, depth=0):
     title_overrides = [i.get('title') for i in nav_tree['contents']]
 
-    def fix_generated_ids(page, content):
-        for i in content.xpath('.//*[starts-with(@id, "auto_")]'):
-            i.attrib['id'] = i.attrib['id'].split('_', 2)[-1]
-        for i in content.xpath('.//*[starts-with(@href, "#auto_")]',
+    # A dictionary to allow look up of a document and new id using the old html
+    # element id
+
+    if id_map is None:
+        id_map = {}
+
+    def fix_generated_ids(page, id_map):
+        """Fix element ids (remove auto marker) and populate id_map."""
+
+        content = content_to_etree(page.content)
+
+        for element in content.xpath('.//*[@id]'):
+            new_ids = []
+            id_val = element.get('id')
+            if id_val.startswith('auto_'):
+                new_val = id_val.split('_', 2)[-1]
+                # Did content from different pages w/ same original id
+                # get moved to the same page?
+                if new_val in new_ids:
```

Comment:
The guts of this conditional block aren't getting hit by any tests. Can you add a test for this?